shoes-swt/spec/shoes/swt/app_spec.rb

```diff
@@ -122,4 +128,20 @@
       it_behaves_like 'reports client area dimensions'
     end
   end
+
```

Comment:
See the comment below for one approach. However mocks are also just fine. You test an object in isolation and otherwise you just test that it sends the correct messages and then that the other object receives and handles those messages correctly. So, here for instance:

one spec:
* stub Shoes.display.getSystemColor to return a well known color (would be different depending on OS otherwise)
* assert that `::Shoes::DSL` receives `define_shoes_color` with the right arguments (rgb, shoes_background_color)

In another test:
* assert that Shoes::DSL.define_shoes_color does what you'd expect it to do given the right inputs

You don't need to explicitly test Shoes.display.getSystemColor as it is a library method :)

Does that help?

You can do the integration style test or the mock or both :)